### PR TITLE
Hooks man

### PR DESCRIPTION
**Pull request by bringking**

### CODE REVIEW

File: hooks.mdx [hunk 1]
```diff
@@ -1 +1,183 @@
...
+  
+  return (
+    <div>
+      Count is: { count }
+      <button onClick={increment}>Increment</button>
+    </div>
+  );
+}
+```
+<SimpleCounter />
```
Comment (theshortcut) [inline hooks.mdx]: omg fake news :trollface:

File: hooks.mdx [hunk 2]
```diff
@@ -1 +1,183 @@
...
+---
+### How does this work?
+---
+```javascript
+// Fake React Internals
```
Comment (theshortcut) [inline hooks.mdx]: nice. saw that tweet yesterday super helpful.

File: hooks.mdx [hunk 3]
```diff
@@ -1 +1,183 @@
...
+---
+* I think classes are/were fine
+* Hooks are compelling for performance and bundle size
+* Could replace "stores" as abstractions for shared business logic
+* We should be careful and fully understand the tradeoffs
```
Comment (theshortcut) [inline hooks.mdx]: worth talking about flattening the react tree?

File: snippets/date-selector-moonshot.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import getDaysInMonth from 'date-fns/get_days_in_month';
+import { rem } from 'polished';
+import { injectIntl } from 'react-intl';
+import { Box, Flex } from '@ghostgroup/grid-styled';
+
+export class DateSelector extends Component {
```
Comment (theshortcut) [inline snippets/date-selector-moonshot.js]: of course it had to be one of my components …

File: hooks.mdx (hunk 1 above)
Comment (bringking) [reply hooks.mdx]: lol, I was looking for a way to show the editable source and running frame with MDX but couldn't find it!

File: snippets/date-selector-moonshot.js (hunk 4 above)
Comment (bringking) [reply snippets/date-selector-moonshot.js]: lol is that one of yours!? nice, well it's a good one cause it's complicated 😄

File: hooks.mdx (hunk 3 above)
Comment (bringking) [reply hooks.mdx]: yeah good call, I will add that in the HOC section

**Pull request merged**